core/state-machine/src/changes_trie/build.rs

```diff
@@ -242,23 +239,23 @@ mod test {
 	#[test]
 	fn build_changes_trie_nodes_on_digest_block_l1() {
 		let (backend, storage, changes) = prepare_for_build();
-		let changes_trie_nodes = prepare_input::<_, _, _, RlpCodec>(&backend, Some(&storage), &changes, 4).unwrap();
+		let changes_trie_nodes = prepare_input(&backend, Some(&storage), &changes, 4).unwrap();
 		assert_eq!(changes_trie_nodes, Some(vec![
 			InputPair::ExtrinsicIndex(ExtrinsicIndex { block: 4, key: vec![100] }, vec![0, 2, 3]),
 			InputPair::ExtrinsicIndex(ExtrinsicIndex { block: 4, key: vec![101] }, vec![1]),
 			InputPair::ExtrinsicIndex(ExtrinsicIndex { block: 4, key: vec![103] }, vec![0, 1]),
 
-			InputPair::DigestIndex(DigestIndex { block: 4, key: vec![100] }, vec![1, 3]),
```

Comment:
@svyatonik I had to change this to make the tests work. any idea why?

Reply:
ok if it's an issue with the iterator, then it would be good to have a test where i can reproduce it...

Reply:
here's the test: https://github.com/paritytech/substrate/commit/93e13a8e1ce54522c2af7eb85c7348a92828fdd6

Reply:
thanks. that's fixed now, but the fix didn't invalidate the changes here...

Reply:
nm false alarm. tests confirm fix.